services/QuillLMS/client/app/entrypoints/snippets/translated_text.ts

```diff
@@ -0,0 +1,78 @@
+function adjustTextareaHeight(textarea) {
+  textarea.style.height = 'auto';
+  textarea.style.height = (textarea.scrollHeight) + 'px';
+}
+
+function initializeTranslatedTexts() {
```

Comment:
This method feels like its doing to many things and probably should be refactored.  I'll be curious to hear what others think.

Reply:
I think I agree.  I'd be inclined to take the code in each of the `if` blocks and make it separate function so that you just end up with a bunch of `if` calls that simply pass `target` through to a function to actually handle the case.